subcommands/keys/ca_create.go

```diff
@@ -73,46 +73,43 @@ func getDeviceCaCommonName(factory string) string {
 func doCreateCA(cmd *cobra.Command, args []string) {
 	factory := viper.GetString("factory")
 	certsDir := args[0]
-	logrus.Debugf("Create CA for %s under %s", factory, certsDir)
+
 	subcommands.DieNotNil(os.Chdir(certsDir))
+	hsm, err := x509.ValidateHsmArgs(
+		hsmModule, hsmPin, hsmTokenLabel, "--hsm-module", "--hsm-pin", "--hsm-token-label")
+	subcommands.DieNotNil(err)
+	x509.InitHsm(hsm)
```

Comment:
As this is the same everywhere, why not extract this into a common collected function and call it with the `args` needed? I just do not know what to call it, maybe like `SetupHSM`? Unsure what the golang conventions are.

Reply:
That was my initial thought.

But, I _believe_ as soon as in Q1 2024 we will allow storing other keys on HSM, like e.g. the device CA key.
And, I'm 100% sure that users would want to store it on a different HSM device than the root CA key.
That led me to understanding that this command is going to grow with other `--hsm-foo` parameters, and we're going to have more than 1 `hsm` local variable in some commands.

I don't want to rewrite a lot of code again at that time; so I kind of prepare it for minimal future changes.

Reply:
I understand. :+1: